src/controllers/userController.js

```diff
@@ -1,28 +1,31 @@
 const User = require('../models/User');
+const jwt = require('jsonwebtoken');
 const { StatusCodes } = require('http-status-codes');
 const { BadRequestError, UnauthenticatedError } = require('../errors');
 
 const register = async (req, res) => {
-    try {
-        const { email } = req.body;
-        
-        const existingUser = await User.findOne({ email });
-        if(existingUser) {
-            return res.status(StatusCodes.BAD_REQUEST).json({ error: 'Invalid credentials' });
-        };
-        
-        const user = await User.create({ ...req.body });
-        const token = user.createJWT();
-
-        const { password, _id, __v, ...userData } = user.toObject();
-        
-        res.status(StatusCodes.CREATED).json({ userData, token });
-    } catch (error) {
-      console.error(error);
-      res.status(StatusCodes.BAD_REQUEST).json({ error: error.message });
-      }
+  try {
+    const { email } = req.body;
```

Comment:
we can check if there is email in the req.body and immediately return a validation check.